src/Temporalio/Bridge/src/metric.rs

```diff
@@ -87,12 +90,19 @@ pub extern "C" fn metric_attributes_free(attrs: *mut MetricAttributes) {
 }
 
 fn metric_attributes_append(
-    orig: &mut metrics::MetricAttributes,
+    meter: &MetricMeter,
+    orig: &metrics::MetricAttributes,
     attrs: *const MetricAttribute,
     size: libc::size_t,
-) {
+) -> MetricAttributes {
     let attrs = unsafe { std::slice::from_raw_parts(attrs, size) };
-    orig.add_new_attrs(attrs.iter().map(metric_attribute_to_key_value));
+    let core = meter.core.inner.extend_attributes(
+        orig.clone(),
+        metrics::NewAttributes {
+            attributes: attrs.iter().map(metric_attribute_to_key_value).collect(),
```

Comment:
Should probably be `into_iter` I think, to avoid an extra clone

Reply:
Don't want to do a move from user-provided/owned slice and I accept a reference in `metric_attribute_to_key_value` and copy everything inside of there as needed.